llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1374,6 +1374,61 @@ LogicalResult TopkOp::generateScalarImplementation(OpBuilder &b, Location loc,
   return success();
 }
 
+SmallVector<unsigned>
+TopkOp::getPartitionableLoops(unsigned maxNumParallelDims) {
+  auto range = llvm::seq<unsigned>(0, getInputRank());
+  SmallVector<unsigned> partitionableLoops(range.begin(), range.end());
+  partitionableLoops.erase(std::next(partitionableLoops.begin(), dimension()));
+  return partitionableLoops;
+}
+
+Operation *TopkOp::getTiledImplementation(OpBuilder &builder,
+                                          ValueRange outputs,
+                                          ArrayRef<OpFoldResult> offsets,
+                                          ArrayRef<OpFoldResult> sizes,
+                                          SmallVectorImpl<Value> &results) {
+  assert(outputs.size() == this->outputs().size());
+  int64_t rank = getInputRank();
+  assert(offsets.size() == static_cast<size_t>(rank) &&
+         sizes.size() == static_cast<size_t>(rank));
+  SmallVector<OpFoldResult> strides(rank, builder.getI64IntegerAttr(1));
+  Location loc = getLoc();
+
+  SmallVector<Value> tiledOperands;
+  tiledOperands.emplace_back(
+      getSlice(builder, loc, values(), offsets, sizes, strides));
+  tiledOperands.emplace_back(
+      getSlice(builder, loc, indices(), offsets, sizes, strides));
+
+  // For output sizes, we need to slice using the K dim of the ouput (tiling
+  // sizes come from the iteration domain based on the input). For Top-K,
+  // the K dim is a reduction and not tiled so we can change sizes[dimension()]
```

Comment:
I am not sure you need to do this. I think the tiling algorithm automatically sets the offsets, sizes, etc. to the untiled value when the loop isnt tiled...
https://github.com/google/iree/blob/5d4d17bc24456375a0f921faefe41a57ad1cdf87/llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Passes/Tiling.cpp#L120

Reply:
I'm trying to illustrate in the comment that the tile sizes provided are based on the input shapes, which are returned  from `getIterationDomain()`.  

>  // For output sizes, we need to slice using the K dim of the output (tiling
>  // sizes come from the iteration domain based on the input).

Even though the tiling size knows this dimension isn't tiled, it provides the untiled input shape here for the `getSlice()` method. This doesn't work because the output will always have a different size for the K dim, so the code is updating the k dim of the slice to come form the `outputValues()` instead of the input, then it slices the output correctly on the following lines.

Reply:
the part that is confusing to me is the "tiling sizes come from the iteration domain based on the input". The iteration domain is what was specified by the op. The tiling algorithm itself has no idea how it relates to the inputs/outputs. I think this comment is much clearer
```
// Replace the tile size for the K dimension to use the output size instead of the input size.
```

Reply:
Done